web/src/app/services/HeartbeatMonitorEditDialog.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from 'react'
+import { useQuery, useMutation, gql } from 'urql'
+import { fieldErrors, nonFieldErrors } from '../util/errutil'
+import FormDialog from '../dialogs/FormDialog'
+import HeartbeatMonitorForm, { Value } from './HeartbeatMonitorForm'
+import Spinner from '../loading/components/Spinner'
+import { GenericError } from '../error-pages'
+
+const mutation = gql`
+  mutation ($input: UpdateHeartbeatMonitorInput!) {
+    updateHeartbeatMonitor(input: $input)
+  }
+`
+const query = gql`
+  query ($id: ID!) {
+    heartbeatMonitor(id: $id) {
+      id
+      name
+      timeoutMinutes
+    }
+  }
+`
+
+export default function HeartbeatMonitorEditDialog(props: {
+  monitorID: string
+  onClose: () => void
+}): JSX.Element {
+  const [value, setValue] = useState<Value | null>(null)
+
+  const [{ data, error, fetching }] = useQuery({
+    query,
+    variables: { id: props.monitorID },
+  })
+  const [updateStatus, update] = useMutation(mutation)
+
+  useEffect(() => {
+    if (!updateStatus.data) return
+    props.onClose()
+  }, [updateStatus.data])
```

Comment:
Should we follow what we do for create and delete and use `.then` /w an error check instead?